staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -825,8 +832,8 @@ func (r *Request) Stream(ctx context.Context) (io.ReadCloser, error) {
 		if err != nil {
 			return nil, err
 		}
-		if r.body != nil {
-			req.Body = io.NopCloser(r.body)
+		if req.Body != nil {
+			req.Body = io.NopCloser(req.Body)
 		}
```

Comment:
maybe we can move this into `r.newHTTPRequest(ctx)` now

Reply:
actually, if the body has a closer don't we want to use it?

Reply:
This is preserving existing behavior, but I don't know why we were wrapping this this way. http.NewRequest already wraps bodies that don't implement closer like this, so the only effect this has previously was to ensure Close on a custom reader body would never be called when streaming (which seems wrong)

Reply:
added in https://github.com/kubernetes/kubernetes/pull/79724/files#diff-e6b16db60c379a7e5fdb8f46054d9a87ea511dba3f94d261b5f61a506fcb5be2R652 before we had a consistent request constructor that set the body. dropping this